### PR TITLE
Feature/create unit tests for message helper

### PR DESCRIPTION
**Pull request by okozlovlux**
Methods:
- CheckWithPolicy 
- CreateAddVRCommandToHMI 
- FillAppRevokedPermissions 
- GetBCActivateAppRequestToHMI 
- GetOnAppInterfaceUnregisteredNotificationToMobile 
- GetPriorityCode 
- MobileResultToString 
- MobileToHMIResult 
- ProcessSoftButtons 
- VerifyImage 
- VerifyImageFiles 

@LevchenkoS @AByzhynar @Kozoriz @vlantonov @anosach-luxoft @wolfylambova22 please review

### CODE REVIEW

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 1]
```diff
@@ -805,6 +814,310 @@ TEST_F(MessageHelperTest,
...
+  EXPECT_EQ((*message)[strings::msg_params][strings::activate_app_hmi_level],
+            hmi_apis::Common_HMILevel::BACKGROUND);
+}
+
+TEST_F(MessageHelperTest, GetBCActivateAppRequestToHMISendPolicyPriorityTrue) {
+  const bool kSendPolicyPriority = true;
+  const uint32_t kCorrelationId = 10u;
+  const uint32_t kAppId = 11u;
+  const std::string kMacAdress = "mac_adress";
+  const std::string& kPolicyAppId = "policy_app_id";
```
Comment (LevchenkoS) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @okozlovlux,` const std::string&`? Are you sure that there must be a reference?

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 2]
```diff
@@ -805,6 +814,310 @@ TEST_F(MessageHelperTest,
...
+  EXPECT_EQ((*message)[strings::params][strings::function_id],
+            mobile_api::FunctionID::OnAppInterfaceUnregisteredID);
+  EXPECT_EQ((*message)[strings::params][strings::message_type],
+            static_cast<int32_t>(kNotification));
+  EXPECT_EQ((*message)[strings::params][strings::connection_key],
+            kConnectionKey);
+  EXPECT_EQ((*message)[strings::msg_params][strings::reason], kReason);
+}
+
+TEST_F(MessageHelperTest, GetPriorityCodeInvalidEnum) {
```
Comment (LevchenkoS) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @okozlovlux, wrong test case naming. Should be like`<method/function name>_<data>_<expected result>`. Please check everywhere.

**Comment by vlantonov** [comment]
Reviewed

**Comment by LevchenkoS** [comment]
@okozlovlux, reviewed.

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 3]
```diff
@@ -805,6 +814,313 @@ TEST_F(MessageHelperTest,
...
+  smart_objects::SmartObject soft_button;
+  soft_button[strings::system_action] = mobile_apis::SystemAction::INVALID_ENUM;
+  message_params[strings::soft_buttons] = soft_button;
+  application_manager_test::MockApplicationManager mock_application_manager;
+  policy_test::MockPolicyHandlerInterface policy_handler;
+
+  EXPECT_CALL(policy_handler, PolicyEnabled()).WillOnce(Return(true));
+  EXPECT_CALL(policy_handler, CheckSystemAction(_, _)).WillOnce(Return(false));
+  EXPECT_CALL(*app_shared_ptr, policy_app_id()).WillOnce(Return(""));
+  EXPECT_EQ(MessageHelper::ProcessSoftButtons(message_params,
```
Comment (AByzhynar) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @okozlovlux EXPECT_EQ(EXPECTED, ACTUAL)

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 4]
```diff
@@ -805,6 +814,313 @@ TEST_F(MessageHelperTest,
...
+                                                policy_handler,
+                                                mock_application_manager),
+              mobile_apis::Result::INVALID_DATA);
+  }
+}
+
+TEST_F(MessageHelperTest, ProcessSoftButtons_Text_InvalidData) {
+  MockApplicationSharedPtr app_shared_ptr =
+      utils::MakeShared<MockApplication>();
+  smart_objects::SmartObject message_params;
```
Comment (AByzhynar) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @okozlovlux I would suggest to declare variables immediately before using.

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 5]
```diff
@@ -805,6 +814,313 @@ TEST_F(MessageHelperTest,
...
+  EXPECT_EQ(MessageHelper::ProcessSoftButtons(message_params,
+                                              app_shared_ptr,
+                                              policy_handler,
+                                              mock_application_manager),
+            mobile_apis::Result::SUCCESS);
+}
+
+TEST_F(MessageHelperTest, CreateAddVRCommandToHMI_ValidVrCommand_SUCCESS) {
+  const uint32_t kCmdId = 1;
+  const smart_objects::SmartObject kVrCommands;
```
Comment (AByzhynar) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @okozlovlux I would suggest to declare variables immediately before using.

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 6]
```diff
@@ -805,6 +814,313 @@ TEST_F(MessageHelperTest,
...
+            (*vr_command)[strings::msg_params][strings::type]);
+  EXPECT_EQ((*valid_vr_command)[strings::params][strings::function_id],
+            (*vr_command)[strings::params][strings::function_id]);
+}
+
+TEST_F(MessageHelperTest, CreateAddVRCommandToHMI_NullAppId_SUCCESS) {
+  const uint32_t kCmdId = 0;
+  const smart_objects::SmartObject kVrCommands;
+  const uint32_t kAppId = 0;
+  const uint32_t kGrammarId = 2;
```
Comment (AByzhynar) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @okozlovlux Adde prefix `u` to all unsigned variables. You did it somewhere, somewhere - not. Keep consistency.

File: src/components/application_manager/test/message_helper/message_helper_test.cc (hunk 3 above)
Comment (okozlovlux) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @AByzhynar fixed 2a2c1fe

File: src/components/application_manager/test/message_helper/message_helper_test.cc (hunk 4 above)
Comment (okozlovlux) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @AByzhynar fixed 2a2c1fe

File: src/components/application_manager/test/message_helper/message_helper_test.cc (hunk 5 above)
Comment (okozlovlux) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @AByzhynar fixed 2a2c1fe

File: src/components/application_manager/test/message_helper/message_helper_test.cc (hunk 6 above)
Comment (okozlovlux) [inline src/components/application_manager/test/message_helper/message_helper_test.cc]: @AByzhynar fixed 2a2c1fe

File: src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc.autosave [hunk 7]
```diff
@@ -0,0 +1,1905 @@
+/*
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc.autosave]: @okozlovlux I believe this file should not be here

**Comment by anosach-luxoft** [comment]
Reviewed

**Comment by wolfylambova22** [comment]
Reviewed.

**Comment by Kozoriz** [comment]
@okozlovlux , can you please tell me : where is cases for MobileResultToString, MobileToHMIResult ?

**Comment by okozlovlux** [comment]
@Kozoriz cases for MobileResultToString, MobileToHMIResult have already been done. See line 1877 message_helper_test.cc

**Comment by okozlovlux** [comment]
Pull request doesn't merge and close, because Travis CI stops build with timeout, even build is successfull.

**Pull request merged**